comment_spell_check/utils/create_checker.py

```diff
@@ -22,13 +24,26 @@ def create_checker(dict_list: list[str] = None) -> spellchecker.SpellChecker:
     english_dict = str(lib_path) + "/resources/en.json.gz"
     logger.info("Loading English dictionary from: %s", english_dict)
     checker.word_frequency.load_dictionary(english_dict)
+    logger.info("number of words: %s", checker.word_frequency.unique_words)
 
     # load the additional dictionaries
     if not isinstance(dict_list, list):
         return checker
     if len(dict_list) > 0:
         for d in dict_list:
             logger.info("Loading additional dictionary from: %s", d)
-            checker.word_frequency.load_text_file(d)
+            if isinstance(d, pathlib.PosixPath):
```

Comment:
The logic here is a bit convoluted, there is no difference between the usage of a pathlib.Path and a string representing the path, but they are separated. This can be simplified: 
```
if isinstance(d, str) and (d.startswith("http://") or d.startswith("https://")):
  response = requests.get(d)
  ... 
else:
  checker.word_frequency.load_text_file(d)
```

Reply:
Here is what I did last time I needed to support both local files and URLs: https://github.com/niaid/rap_sitkCore/blob/b43d84054c116c64a94d2188a9fa066b4562a710/rap_sitkcore/read_dcm_headers.py#L133-L138

The python conventions is that it is easier to ask for forgiveness than permission (EAFP): https://realpython.com/python-lbyl-vs-eafp/ 

In this ca it means try it as a URL with requests, and if it fails that try it as a Path.